File: CPP/Casing.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

// Kelas Casing yang mewarisi dari kelas Komponen
class Casing : public Komponen {
    private:
        string tipeCasing;
    
    public:
        Casing() {}

        // Konstruktor dengan parameter untuk menginisialisasi tipeCasing, merk, dan nama
        Casing(string tipeCasing, string merk, string nama) 
            : Komponen(merk, nama) {
            this->tipeCasing = tipeCasing;
        }
    
        // Setter
        void setTipeCasing(string tipeCasing) {
            this->tipeCasing = tipeCasing;
        }
        
        // Getter
        string getTipeCasing() {
            return tipeCasing;
        }
    
    ~Casing()
    {
        
    }
};

File: python/Casing.py
from Komponen import Komponen

class Casing(Komponen):
    #atribut private
    __tipe_casing = ""

    #constructor
    def __init__(self, tipe_casing, merk, nama):
        super().__init__(merk, nama)
        self._tipe_casing = tipe_casing

    #setter
    def set_tipe_casing(self, tipe_casing):
        self._tipe_casing = tipe_casing

    #getter
    def get_tipe_casing(self):
        return self._tipe_casing